Sming/Components/Partition/src/include/Partition.h

```diff
@@ -0,0 +1,227 @@
+	XX(Ota0, ESP_PARTITION_SUBTYPE_APP_OTA_0, "OTA #0")                                                                \
+	XX(Ota1, ESP_PARTITION_SUBTYPE_APP_OTA_1, "OTA #1")                                                                \
+	XX(Ota2, ESP_PARTITION_SUBTYPE_APP_OTA_2, "OTA #2")                                                                \
+	XX(Ota3, ESP_PARTITION_SUBTYPE_APP_OTA_3, "OTA #3")                                                                \
+	XX(Ota4, ESP_PARTITION_SUBTYPE_APP_OTA_4, "OTA #4")                                                                \
+	XX(Ota5, ESP_PARTITION_SUBTYPE_APP_OTA_5, "OTA #5")                                                                \
+	XX(Ota6, ESP_PARTITION_SUBTYPE_APP_OTA_6, "OTA #6")                                                                \
+	XX(Ota7, ESP_PARTITION_SUBTYPE_APP_OTA_7, "OTA #7")                                                                \
+	XX(Ota8, ESP_PARTITION_SUBTYPE_APP_OTA_8, "OTA #8")                                                                \
+	XX(Ota9, ESP_PARTITION_SUBTYPE_APP_OTA_9, "OTA #9")                                                                \
+	XX(Ota10, ESP_PARTITION_SUBTYPE_APP_OTA_10, "OTA #10")                                                             \
+	XX(Ota11, ESP_PARTITION_SUBTYPE_APP_OTA_11, "OTA #11")                                                             \
+	XX(Ota12, ESP_PARTITION_SUBTYPE_APP_OTA_12, "OTA #12")                                                             \
+	XX(Ota13, ESP_PARTITION_SUBTYPE_APP_OTA_13, "OTA #13")                                                             \
+	XX(Ota14, ESP_PARTITION_SUBTYPE_APP_OTA_14, "OTA #14")                                                             \
+	XX(Ota15, ESP_PARTITION_SUBTYPE_APP_OTA_15, "OTA #15")                                                             \
+	XX(Test, ESP_PARTITION_SUBTYPE_APP_TEST, "Test application")
+
+#define PARTITION_DATA_TYPE_MAP(XX)                                                                                    \
+	XX(Ota, ESP_PARTITION_SUBTYPE_DATA_OTA, "OTA selection")                                                           \
+	XX(Phy, ESP_PARTITION_SUBTYPE_DATA_PHY, "PHY init data")                                                           \
+	XX(Nvs, ESP_PARTITION_SUBTYPE_DATA_NVS, "NVS")                                                                     \
+	XX(CoreDump, ESP_PARTITION_SUBTYPE_DATA_COREDUMP, "Core Dump data")                                                \
+	XX(NvsKeys, ESP_PARTITION_SUBTYPE_DATA_NVS_KEYS, "NVS key information")                                            \
+	XX(EfuseEm, ESP_PARTITION_SUBTYPE_DATA_EFUSE_EM, "eFuse emualtion")                                                \
+	XX(EspHttpd, ESP_PARTITION_SUBTYPE_DATA_ESPHTTPD, "ESPHTTPD")                                                      \
```

Comment:
> XX(EspHttpd, ESP_PARTITION_SUBTYPE_DATA_ESPHTTPD, "ESPHTTPD")

Do we need this?

Reply:
No idea, this is simply laying out the supported esp32 partition types. If someone wants to use it then it should be there, but at the very least it's a 'reserved' type.